packages/@guardian/source-foundations/src/typography/font-styles.ts

```diff
@@ -0,0 +1,89 @@
+import {
+	fonts,
+	fontWeights,
+	fontWeightsAvailable,
+	italicsAvailableForFontWeight,
+	lineHeights,
+	pxTextSizes,
+	remTextSizes,
+	underlineThickness,
+} from './data';
+import type {
+	Categories,
+	FontScaleArgs,
+	FontStyle,
+	Option,
+	TypographyOptions,
+	TypographyStyles,
+} from './types';
+
+const determineFontStyleProperty = (
+	fontStyle: Option<FontStyle>,
+	hasItalic: boolean,
+) => {
+	switch (fontStyle) {
+		case 'italic':
+			return hasItalic ? 'italic' : undefined;
+		case 'normal':
+			return 'normal';
+		case null:
+		default:
+			return undefined;
+	}
+};
+
+export const fontStyleFunction =
+	<
+		Category extends keyof Categories,
+		Level extends keyof Categories[Category],
+	>(
+		category: Category,
+		level: Level,
+		defaults: TypographyOptions,
+	) =>
+	(options?: FontScaleArgs): TypographyStyles => {
+		const finalOptions = {
```

Comment:
could this be something like `fontSpec` or similar?

Reply:
i've put in a change to `finalFontConfiguration`, let me know if that's clearer - or if `spec` (i'm guessing for "specification"?) would be clearer

Reply:
Yeah i think thats fine.